File: src/php/Rpc/Services/RpcModuleIf.php
<?php
/**
 * Created by PhpStorm.
 * User: zhangyubo
 * Date: 2019/4/30
 * Time: 15:21
 */

namespace Cecd\Sdk\Rpc\Services;


interface RpcModuleIf
{
    public function getHost();

    public function getPort();

    public function setHost($host);

    public function setPort($port);
}

File: src/java/src/main/java/com/xmi01/thriftrpc/rpc/RpcModuleIf.java
package com.xmi01.thriftrpc.rpc;

import com.xmi01.thriftrpc.rpc.interceptor.ClientInterceptor;

public interface RpcModuleIf {

    public int getServiceId();

    public String getServiceName();

    public String getHost();

    public RpcModuleIf setHost(String host);

    public int getPort();

    public RpcModuleIf setPort(int port);

    public String getLang();

    public int getTimeout();

    public RpcModuleIf setTimeout(int timeout);

    public void setInterceptor(ClientInterceptor interceptor);

    public ClientInterceptor getInterceptor();

    public void setDebug(boolean debug);

    public boolean getDebug();
}
